src/dragonfly_reporter/dependencies/microsoft_graph_dependencies.py

```diff
@@ -0,0 +1,16 @@
+"""Microsoft Graph integration."""
+
+from azure.identity.aio import ClientSecretCredential
+from msgraph import GraphServiceClient
+
+from dragonfly_reporter.constants import Microsoft
+
+
+def build_graph_client() -> GraphServiceClient:
```

Comment:
Cache using `@functools.cache` so we don't create a new client for every request

Reply:
This will fail because of https://github.com/Azure/azure-sdk-for-python/issues/32309